src/cmd/CMakeLists.txt

```diff
@@ -1,3 +1,86 @@
+# Collect source files into the "sources" variable and unit test files into the
+# "gtest_sources" variable.
+gz_get_libsources_and_unittests(sources gtest_sources)
+
+add_library(gz STATIC gz.cc)
+target_link_libraries(gz
+  PUBLIC
+    ${PROJECT_LIBRARY_TARGET_NAME}
+)
+
+set(model_executable gz-sim-model)
+add_executable(${model_executable} model_main.cc)
+target_link_libraries(${model_executable}
+  gz
+  gz-utils${GZ_UTILS_VER}::cli
+  ${PROJECT_LIBRARY_TARGET_NAME}
+)
+
+install(
+  TARGETS ${model_executable}
+  DESTINATION ${CMAKE_INSTALL_LIBEXECDIR}/gz/${GZ_DESIGNATION}${PROJECT_VERSION_MAJOR}/)
+
+# Build the unit tests.
+set(test_sources)
+
+# gz_TEST and ModelCommandAPI_TEST are not supported with multi config
+# CMake generators, see also cmd/CMakeLists.txt
+get_property(GENERATOR_IS_MULTI_CONFIG GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+if(NOT GENERATOR_IS_MULTI_CONFIG)
+  list(APPEND test_sources
+    ../gz_TEST.cc
+    ../ModelCommandAPI_TEST.cc
+  )
+endif()
+
+gz_build_tests(TYPE UNIT
+  SOURCES
+    ${test_sources}
+  LIB_DEPS
+    gz-utils${GZ_UTILS_VER}::gz-utils${GZ_UTILS_VER}
+    ${EXTRA_TEST_LIB_DEPS}
+    gz-sim${PROJECT_VERSION_MAJOR}
+  ENVIRONMENT
+      GZ_SIM_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}
+)
+
+foreach(CMD_TEST
+  UNIT_gz_TEST UNIT_ModelCommandAPI_TEST)
+
+  if(NOT TARGET ${CMD_TEST})
+    continue()
+  endif()
+
+  # Running `gz sim` on macOS has problems when run with /usr/bin/ruby
+  # due to System Integrity Protection (SIP). Try to find ruby from
+  # homebrew as a workaround.
+  if (APPLE)
+    find_program(BREW_RUBY ruby HINTS /usr/local/opt/ruby/bin)
+  endif()
```

Comment:
If we're running just `UNIT_ModelCommandAPI_TEST`, we shouldn't have to worry about this since we won't be loading dlls.